docs/source/en/tasks_explained.mdx

```diff
@@ -0,0 +1,230 @@
+specific language governing permissions and limitations under the License.
+-->
+
+# How 🤗 Transformers solve tasks
+
+In [What 🤗 Transformers can do](task_summary), you learned about natural language processing (NLP), speech and audio, computer vision tasks, and some important applications of them.
+
+This page will take a closer look at how models solve these tasks and explain what's going on. There are many ways to solve a given task, some models may implement certain techniques or even approach the task from a new angle, but for Transformer models, the general idea is the same. Owing to its flexible architecture, most models are a variant of an encoder, decoder, or encoder-decoder structure. The exception is convolutional neural networks (CNNs) which are still used today for computer vision tasks, but we'll also explain how these work.
+
+To keep things relatively brief, we'll provide a high-level overview of how each task is solved. The models we'll look at are among the most popular ones:
+
+- [wav2vec2](model_doc/wav2vec2) for audio classification and automatic speech recognition (ASR)
+- [Vision Transformer (ViT)](model_doc/vit) and [ConvNeXT](model_doc/convnext) for image classification
+- [DETR](model_doc/detr) for image segmentation and object detection
+- [GLPN](model_doc/glpn) for depth estimation
+- [BERT](model_doc/bert) for NLP tasks like text classification, token classification and question answering that use an encoder
+- [GPT2](model_doc/gpt2) for NLP tasks like text generation that use a decoder
+- [BART](model_doc/bart) for NLP tasks like summarization and translation that use an encoder-decoder
+
+<Tip>
+
+Before you go further, it is good to have a basic understanding of the original Transformer architecture. Knowing how encoders, decoders, and attention work would really help. If you're just getting started or need a refresher, check out our [course](https://huggingface.co/course/chapter1/4?fw=pt) for more information about how Transformers work! 
+
+</Tip>
+
+## Speech and audio
+
+[wav2vec2](model_doc/wav2vec2) is a self-supervised model pretrained on unlabeled speech data and finetuned on labeled data for audio classification and automatic speech recognition. This model has four main components:
+
+1. A *feature encoder* takes the raw waveform, normalizes it to zero mean and unit variance, and converts it into a sequence of feature vectors that are each 20ms long.
+
+2. Next, the feature vectors are passed to a *quantization module*. The quantizer's goal is to learn discrete speech units. Audio waveforms are continuous by nature, so they can't be broken up into separate units like a sequence of text can be split into words. So what are speech units? Each speech unit is a speech representation sampled from a group of *codebooks*, and each codebook contains a set vocabulary. wav2vec2 has two codebooks with 320 codewords in each codebook for a total of 102,400 speech units.
```

Comment:
I tried to explain this a little better since I think it's important for the user to understand where the input is coming from. Maybe one of our audio experts can help give a clearer explanation 😅